From jejb@kernel.org  Thu Oct 16 15:33:09 2008
From: Cornelia Huck <cornelia.huck@de.ibm.com>
Date: Thu, 16 Oct 2008 22:05:05 GMT
Subject: Driver core: Fix cleanup in device_create_vargs().
To: jejb@kernel.org, stable@kernel.org
Message-ID: <200810162205.m9GM55pF005776@hera.kernel.org>

From: Cornelia Huck <cornelia.huck@de.ibm.com>

commit 286661b3777897220ecfcd774bccc68a34667f39 upstream

If device_register() in device_create_vargs() fails, the device
must be cleaned up with put_device() (which is also fine on NULL)
instead of kfree().

Signed-off-by: Cornelia Huck <cornelia.huck@de.ibm.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/base/core.c
+++ b/drivers/base/core.c
@@ -1243,7 +1243,7 @@ struct device *device_create_vargs(struc
 	return dev;
 
 error:
-	kfree(dev);
+	put_device(dev);
 	return ERR_PTR(retval);
 }
 EXPORT_SYMBOL_GPL(device_create_vargs);
